### ISSUE TITLE
Speed

### ISSUE THREAD

**Issue by MisterTeo** [brichard19/BitCrack#232]
I tried many setups but this is the best speed on gtx 1650. Someone with other advices ? Built in RELEASE mode. Tried in DEBUG mod, speed is less...
![2020-04-11 11_53_37-Window](https://user-images.githubusercontent.com/34290070/79039680-a096f480-7beb-11ea-8720-8579d802f7c9.png)

**Comment by RastaDollo** [comment]
Try -b 896 -t 32 -p 890

**Comment by MisterTeo** [comment]
![2020-04-30 19_08_45-Window](https://user-images.githubusercontent.com/34290070/80733307-33a1bb00-8b16-11ea-9d28-a35fdf1cdfaf.png)
Thank you RastDollo, but I have lower speed with this settingg...

**Comment by djarumlights** [comment]
I just did a fork that changes how default parameters are set, memory usage, etc (basically combined a few other people's optimizations into a new release). Maybe give it a try? It's a bit faster than the original version on my 1080 mini (went from 240-250 up to 270 MK/s). 

Also, I see you're doing both compressed and uncompressed. Uncompressed is slower overall, so it may be bogging you down a bit.

**Comment by MisterTeo** [comment]
Yes @djarumlights, I do both, compressed and uncompressed. I tried only compressed and I had a better speed ( around 225 Mkeys ) but I do both because I don't know how are the keys in my list, compressed or uncompressed. 
Now I switched to debian linux, speed is less then windows ( 93 Mkeys ). 
Thank you for reply.

**Comment by djarumlights** [comment]
https://github.com/djarumlights/BitCrack/releases/tag/0.32a

I have it running now without any issues overnight. It's literally some copy/paste work putting the work of people far smarter/more capable than I together. Hope it helps!

**Comment by MisterTeo** [comment]
make[1]: Entering directory '/home/teo/Downloads/BitCrack-0.32a/KeyFinder'
g++ -DBUILD_CUDA -o cuKeyFinder.bin ConfigFile.cpp main.cpp DeviceManager.cpp  -I/home/teo/Downloads/BitCrack-0.32a/util  -I/home/teo/Downloads/BitCrack-0.32a/AddressUtil  -I/home/teo/Downloads/BitCrack-0.32a/CmdParse  -I/home/teo/Downloads/BitCrack-0.32a/CryptoUtil  -I/home/teo/Downloads/BitCrack-0.32a/KeyFinderLib  -I/home/teo/Downloads/BitCrack-0.32a/CLKeySearchDevice  -I/home/teo/Downloads/BitCrack-0.32a/CudaKeySearchDevice  -I/home/teo/Downloads/BitCrack-0.32a/cudaMath  -I/home/teo/Downloads/BitCrack-0.32a/clUtil  -I/home/teo/Downloads/BitCrack-0.32a/cudaUtil  -I/home/teo/Downloads/BitCrack-0.32a/secp256k1lib  -I/home/teo/Downloads/BitCrack-0.32a/Logger  -I/home/teo/Downloads/BitCrack-0.32a/embedcl -I/usr/local/cuda/include -O2 -std=c++11 -L/home/teo/Downloads/BitCrack-0.32a/lib -L/usr/local/cuda/lib64 -lkeyfinder -laddressutil -lsecp256k1 -lcryptoutil -lsecp256k1 -lcudautil -llogger -lutil -lCudaKeySearchDevice -lcudadevrt -lcudart -lcmdparse
main.cpp: In function ‘DeviceParameters getDefaultParameters(const DeviceManager::DeviceInfo&)’:
main.cpp:235:35: error: ‘log’ was not declared in this scope
  p.pointsPerThread = pow(2, floor(log(device.memory / (4 * 1024 * p.blocks)) / log(2))); //best power of 2 based on device memory assuming 4kb per block.
                                   ^~~
main.cpp:235:35: note: suggested alternative: ‘long’
  p.pointsPerThread = pow(2, floor(log(device.memory / (4 * 1024 * p.blocks)) / log(2))); //best power of 2 based on device memory assuming 4kb per block.
                                   ^~~
                                   long
main.cpp:235:29: error: ‘floor’ was not declared in this scope
  p.pointsPerThread = pow(2, floor(log(device.memory / (4 * 1024 * p.blocks)) / log(2))); //best power of 2 based on device memory assuming 4kb per block.
                             ^~~~~
main.cpp:235:29: note: suggested alternative: ‘float’
  p.pointsPerThread = pow(2, floor(log(device.memory / (4 * 1024 * p.blocks)) / log(2))); //best power of 2 based on device memory assuming 4kb per block.
                             ^~~~~
                             float
main.cpp:235:22: error: ‘pow’ was not declared in this scope
  p.pointsPerThread = pow(2, floor(log(device.memory / (4 * 1024 * p.blocks)) / log(2))); //best power of 2 based on device memory assuming 4kb per block.
                      ^~~
main.cpp:235:22: note: suggested alternative: ‘putw’
  p.pointsPerThread = pow(2, floor(log(device.memory / (4 * 1024 * p.blocks)) / log(2))); //best power of 2 based on device memory assuming 4kb per block.
                      ^~~
                      putw
make[1]: *** [Makefile:5: all] Error 1
make[1]: Leaving directory '/home/teo/Downloads/BitCrack-0.32a/KeyFinder'
make: *** [Makefile:90: dir_keyfinder] Error 2
teo@debian:~/Downloads/BitCrack-0.32a$ 

Error building...

**Comment by MisterTeo** [comment]
Original version, changed to cuda 10.2, no error building cuda and opencl

**Comment by djarumlights** [comment]
Ok great. Must have missed the linux side for changing to 10.2 (I'm flying by seat of pants here in putting it together). I had to smack it around a few times to build for Windows. 

I didn't do any of the OpenCL optimizations floating around due to having no AMD cards to test with. I'm sure some enterprising soul can slap those into a fork and see what happens. Is it working better for you or nah?

**Comment by MisterTeo** [comment]
I didn't manage to compile on debian, so I can't give my opinion ... maybe when and if I try on windows...

**Comment by djarumlights** [comment]
Ahhh I see. I put up an already-compiled one for Windows if you like. Check the releases tab.

**Comment by MisterTeo** [comment]
@djarumlights I build it on windows...works like a charm. 170 Mkeys vs 103 Mkeys .I have to say, I don't use list of bitcoin addresses, I started just one address, anyway, is faster your version with at least 25 Mkeys.
Works random version to... 167 Mkeys with one address...
Still slower, maybe is my graphic card the problem ( is new ) but faster then original version.

**Comment by djarumlights** [comment]
Long as it's working for someone besides me! Might tinker with -b -t and -p to see if you can bump it up a hair. I use -b 40 -t 512 -p 512 on this 1080 to get 250Mk/s. Glad I could be a little help!

**Comment by MisterTeo** [comment]
With your config I get 150 Mkeys...I guess 170 is maximum...I tried so manny ways, this is it..
